### ISSUE TITLE
Comparability of unions of literal numbers is too strict

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#10119]
On master:

```ts
function f(onethree: 1 | 3, two: 2) {
  return onethree < two;
}
```

Expected: No error.
Actual: Operator '<' cannot be applied to types '1 | 3' and '2'.

**Comment by mhegazy** [comment]
closing in favor of #10120.
